app/forms/articles/destroy_form.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module Articles
+  class DestroyForm < Articles::BaseForm
+    def save
+      ActiveRecord::Base.transaction do
+        save_article_tags!
+        @article.destroy!
+        check_and_destroy_tags!
+        raise ActiveRecord::Rollback unless errors.empty?
+      end
+
+      errors.empty?
+    end
+
+    private
+
+    def save_article_tags!
+      @article_tags_list = @article.tags
+    end
+
+    def check_and_destroy_tags!
+      @article_tags_list.each do |tag|
+        tag.destroy! if tag.present? && tag.articles.count < 1
```

Comment:
do you really need `tag.present?`
`tag.articles.count < 1 ` => '!tag.articles.exists?'

Reply:
It was necessary to change the logic of the destroy form. When I initially destroyed the article, the form exited without getting to the tags. Now I process the tags first and finish by destroying the article.